conn.go

```diff
@@ -584,6 +617,60 @@ type callReq struct {
 	timer *time.Timer
 }
 
+type writeCoalescer struct {
+	w       io.Writer
+	timeout time.Duration
+
+	cond    *sync.Cond
+	mu      sync.Mutex
+	buffers net.Buffers
+
+	// result of the write
+	err error
+}
+
+func (w *writeCoalescer) flush() {
+	if w.timeout > 0 {
+		type deadliner interface {
+			SetWriteDeadline(time.Time) error
+		}
+		w.w.(deadliner).SetWriteDeadline(time.Now().Add(w.timeout))
+	}
+
+	defer w.mu.Unlock()
```

Comment:
very small nitpick: would you mind swapping the order of Unlock() and Lock() here? I understand it's doing the right thing, it just looks weird ;-)